### PR TITLE
web: add telemetry api for embedders

### PR DESCRIPTION
**Pull request by connor4312**
This exports publicLog2. I want to use this on the vscode.dev embedder side to record performance information about initialization times related to Tanha and I's ongoing work.

### CODE REVIEW

**Comment by bpasero** [comment]
Can't this be done from an extension too? Should we be using our perf framework? 

Btw, even if we export it, why `publicLog2` as a name 😱

**Comment by connor4312** [comment]
> Can't this be done from an extension too? Should we be using our perf framework?

not here -- extensions don't get activated before the connection to the remote is established, and we want to be able to collect telemetry about the remote connection failing.

> Btw, even if we export it, why publicLog2 as a name 😱

For the [code annotation extractor](https://github.com/microsoft/vscode-telemetry-extractor/blob/main/documentation/typescript-code-annotations.md) 😬

**Comment by bpasero** [comment]
> extensions don't get activated before the connection to the remote is established

yeah sure, I would have suggested to add `perf.` calls from the HTML page that get's loaded on vscode.dev and collect that later.

**Comment by connor4312** [comment]
Following discussion with Logan, we now only expose the telemetry level (also added a way to watch for changes) and will use app insights directly in the browser.

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/web.api.ts [hunk 1]
```diff
@@ -3,18 +3,20 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import type { PerformanceMark } from 'vs/base/common/performance';
-import type { UriComponents, URI } from 'vs/base/common/uri';
-import type { IWebSocketFactory } from 'vs/platform/remote/browser/browserSocketFactory';
-import type { IURLCallbackProvider } from 'vs/workbench/services/url/browser/urlService';
-import type { LogLevel } from 'vs/platform/log/common/log';
-import type { IUpdateProvider } from 'vs/workbench/services/update/browser/updateService';
 import type { Event } from 'vs/base/common/event';
```
[the remark this answers is not in the document]
Comment (connor4312) [reply src/vs/workbench/browser/web.api.ts:6]: Oop. I sort import when I modify them by force of habit, should I revert the sorting here?

[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/workbench/browser/web.api.ts:6]: Yeah I think sorting imports does not contribute to the PR.

File: src/vs/workbench/browser/web.api.ts [hunk 2]
```diff
@@ -62,6 +64,11 @@ export interface IWorkbench {
 		 * workbench.
 		 */
 		openUri(target: URI): Promise<boolean>;
+
+		/**
+		 * Current workbench telemetry level.
+		 */
+		telemetryLevel: IObservableValue<TelemetryLevel>;
```
Comment (bpasero) [inline src/vs/workbench/browser/web.api.ts:71]: Why is this not a `readonly` getter, but a function?

File: src/vs/workbench/browser/web.factory.ts [hunk 3]
```diff
@@ -120,6 +122,12 @@ export namespace env {
 
 		return workbench.env.openUri(target);
 	}
+
+	export async function telemetryLevel(): Promise<IObservableValue<TelemetryLevel>> {
```
Comment (bpasero) [inline src/vs/workbench/browser/web.factory.ts:126]: Suggest a readonly getter.

**Review by bpasero — CHANGES_REQUESTED** [review]

**Review by bpasero — CHANGES_REQUESTED** [review]
Build failures it seems.

**Review by bpasero — APPROVED** [review]

**Pull request merged**